src/e2e/java/teammates/e2e/cases/e2e/BaseE2ETestCase.java

```diff
@@ -146,4 +153,49 @@ protected AdminHomePage loginAdmin() {
         return loginAdminToPage(createUrl(Const.WebPageURIs.ADMIN_HOME_PAGE), AdminHomePage.class);
     }
 
+    /**
+     * Deletes file with fileName from the downloads folder.
+     */
+    protected void deleteDownloadsFile(String fileName) {
+        String filePath = getTestDownloadsFolder() + fileName;
+        FileHelper.deleteFile(filePath);
+    }
+
+    /**
+     * Verifies downloaded file has correct fileName and contains expected content.
+     */
+    protected void verifyDownloadedFile(String expectedContent, String expectedFileName) {
+        try {
+            String filePath = getTestDownloadsFolder() + expectedFileName;
+            assertTrue(FileHelper.readFile(filePath).contains(expectedContent));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Verifies that email with subject is found in inbox.
+     * Email used must be an authentic gmail account.
+     */
+    protected void verifyEmailSent(String email, String subject) {
+        if (TestProperties.isDevServer()) {
+            return;
+        }
+        EmailAccount emailAccount = new EmailAccount(email);
+        try {
+            emailAccount.getUserAuthenticated();
+            int retryLimit = 5;
+            boolean actual = emailAccount.isEmailWithSubjectPresent(subject);
+            while (!actual && retryLimit > 0) {
+                retryLimit--;
+                ThreadHelper.waitFor(1000);
+                actual = emailAccount.isEmailWithSubjectPresent(subject);
```

Comment:
I have seen a lot of this kind of operation, and it's not just on this PR. Is it possible to build some kind of abstraction for them? Or perhaps use the `RetryManager` class? But don't need to do on this PR.